open_spiel/algorithms/external_sampling_mccfr.cc

```diff
@@ -35,8 +52,15 @@ ExternalSamplingMCCFRSolver::ExternalSamplingMCCFRSolver(const Game& game,
 ExternalSamplingMCCFRSolver::ExternalSamplingMCCFRSolver(
     const Game& game, std::shared_ptr<Policy> default_policy, int seed,
     AverageType avg_type)
-    : game_(game.Clone()),
-      rng_(new std::mt19937(seed)),
+    : ExternalSamplingMCCFRSolver(game.Clone(), default_policy,
+                                  std::make_unique<std::mt19937>(seed),
+                                  avg_type) {}
+
+ExternalSamplingMCCFRSolver::ExternalSamplingMCCFRSolver(
+    std::shared_ptr<const Game> game, std::shared_ptr<Policy> default_policy,
+    std::unique_ptr<std::mt19937> rng, AverageType avg_type)
```

Comment:
Why pass in the RNG rather than the seed? Passing in the complex object here has the problem of forcing one to use std::mt19937 but also making it harder to expose to python.

Reply:
This constructor is only meant for deserialization purposes, i.e. having only the seed is not sufficient to reconstruct the RNG if it was already sampled from. For the same reason, this function should not be exposed to Python as this is already done implicitly via pickling. As above, please let me know if I should expose setters rather than create "deserialization" constructors.